### DISCUSSION TITLE
Better way to purge unused thumbs?

### DISCUSSION THREAD

**Discussion by pinnick** [pocketbase/pocketbase#4755] [Q&A] [answered]
When a thumb is lazily generated for a specific size (say 1000x500) and the custom thumb for the database collection is changed (to say 1000x800), all of the previously generated thumbs of 1000x500 remain in the database. This waste of storage can be particularly annoying for large thumbs. 

The only efficient way I found to get rid of these unused images is to use Linux commands (I believe `find . -name "*1000x500*" -delete`). Does Pocketbase have a feature to delete these images? I feel as though there should be a button to purge unused thumbs.

**Comment by ganigeorgiev** [comment, marked as answer]
Thumbs are not stored in the database, they are stored only as files in the local filesystem or S3.

There is no builtin command at the moment that will delete old thumbs (_all thumbs should be deleted automatically after the file or record deletion_).

**Reply by pinnick** [reply]
Could I try to implement?

**Reply by ganigeorgiev** [reply]
See https://github.com/pocketbase/pocketbase/discussions/4355 _(note that this particular operation can be implemented as external command via [Go](https://pocketbase.io/docs/go-console-commands/) or [JS](https://pocketbase.io/docs/js-console-commands/))_.
